### PR TITLE
True theme

### LINKED ISSUE #544: 建议用户暂时不升级至 V0.5.2
### 建议用户慎重升级 V0.5.2 ，该版本默认颜色会和以前有较大的偏差，可能会引起不适

在 v0.5.2 版本中，升级了自定义主题配置功能，但是这也带来一些后续的问题

如 Pie 图中，升级到 v0.5.2 默认效果

```python
def test_pie_default():
    v1 = [11, 12, 13, 10, 10, 10]
    pie = Pie("饼图示例")
    pie.add("", CLOTHES, v1, is_label_show=True)
    pie.render()

test_pie_default()
```

image

这时候可以使用  Style 类来解决这个问题，

```python
from pyecharts import Pie, Style
style = Style(
    title_color="black",
    subtitle_color="#aaa"
)
pie_style = style.add(
    label_text_color="black",
    legend_text_color="black",
)
# or anyother color you like

def test_pie_default():
    v1 = [11, 12, 13, 10, 10, 10]
    pie = Pie("饼图示例", **style.init_style)
    pie.add("", CLOTHES, v1, is_label_show=True, **pie_style)
    pie.render()

test_pie_default()
```

效果

image

### PR DESCRIPTION
**Pull request by chfw**
fix #544 

vintage:

screen shot 2018-05-09 at 08 22 50

default:

screen shot 2018-05-09 at 08 23 53

### CODE REVIEW

File: pyecharts/utils/__init__.py [hunk 1]
```diff
@@ -89,3 +91,33 @@ def _add(_item):
...
     return front_required_items + fol + dependencies
+
+
+def _expand(dict_generator):
+    return dict(list(dict_generator))
+
+
+def _clean(mydict):
+    for key, value in viewitems(mydict):
+        if value is not None:
```
Comment (chenjiandongx) [inline pyecharts/utils/__init__.py]: 我觉得应该考虑加入 key 值判断，目前主要是处理 key 值包含 color 字符的情况，其他情况我认为可以忽略

File: pyecharts/utils/__init__.py [hunk 2]
```diff
@@ -89,3 +91,33 @@ def _add(_item):
...
+            if isinstance(value, dict):
+                value = _expand(_clean(value))
+            elif isinstance(value, list):
```
Comment (chenjiandongx) [inline pyecharts/utils/__init__.py]: tuple 考虑一下？

**Comment by chfw** [comment]
None, {} and ""(empty string) are cleaned.

**Comment by chenjiandongx** [comment]
> None, {} and ""(empty string) are cleaned.

@chfw 这个是出于什么考虑

**Comment by chfw** [comment]
Name: ""

textStyle: {}

**Comment by kinegratii** [comment]
@chfw  You may distinguish **false values** and **empty values**.

For example ``0``, ``Decimal(0)``, ``time(0)`` are non-empty values.

In Django,this is what data is validated.

```python

EMPTY_VALUES = (None, '', [], (), {})
if v in EMPTY_VALUES:
    pass
```
See  https://github.com/django/django/blob/3574a6d32fcd88d404b110a8d2204db1dd14a545/django/core/validators.py#L13

**Comment by chfw** [comment]
yeah, I noticed it. I am having another fix coming a bit later

File: pyecharts/utils/__init__.py (hunk 1 above)
Comment (chfw) [reply pyecharts/utils/__init__.py]: 在这个 PR 里，你会看到很多 null 的 key.

**Comment by chfw** [comment]
python 2.7 does not like: value not in (None, ''...)

```
FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  if value not in (None, '', [], (), {})
```

File: test/test_utils.py [hunk 3]
```diff
@@ -89,3 +90,30 @@ def test_lazy_object():
...
+                "normal": 1,
+                "empty_string": "",
+                "empty_array": [],
+                "empty_nested_dict": {},
+            },
+        ],
+    }
+    actual_result = remove_key_with_none_value(fixture)
+    expected = {
+        'a': 1,
```
Comment (chenjiandongx) [inline test/test_utils.py]: 为什么 white 了之后单引号和双引号还是没有统一

File: pyecharts/utils/__init__.py [hunk 4]
```diff
@@ -89,3 +100,51 @@ def _add(_item):
             _add(_d)
     fol = [x for x in front_optional_items if x in fist_items]
     return front_required_items + fol + dependencies
+
+
+def _expand(dict_generator):
+    return dict(list(dict_generator))
+
+
+def _clean(mydict):
```
Comment (chenjiandongx) [inline pyecharts/utils/__init__.py]: 函数名称改为 _clean_dict 是否更好一些

File: make.bat [hunk 5]
```diff
@@ -1,2 +1,2 @@
 cd test
-nosetests --with-coverage --cover-package pyecharts --cover-package test && cd .. && flake8 --exclude docs --builtins=unicode,xrange,long
\ No newline at end of file
+nosetests --with-coverage --cover-package pyecharts --cover-package test && cd .. && flake8 --exclude docs --builtins=unicode,xrange,long,basestring
```
Comment (chenjiandongx) [inline make.bat]: flake8 --max-complexity 12

Comment (chfw) [reply make.bat]: done

File: pyecharts/utils/__init__.py (hunk 4 above)
Comment (chfw) [reply pyecharts/utils/__init__.py]: done

**Comment by chenjiandongx** [comment]
@chfw you need to see this https://github.com/pyecharts/pyecharts/blob/7ff921af5e2bd448789ed60dc011b319db9436f8/docs/zh-cn/themes.md. awesome!!!

**Pull request merged**